build/Targets/Settings.props

```diff
@@ -71,6 +71,9 @@
 
     <RoslynNetSdkRootPath>$(MSBuildSDKsPath)\Microsoft.NET.Sdk\Sdk\</RoslynNetSdkRootPath> 
     <RestoreProjectStyle>PackageReference</RestoreProjectStyle>
+
+    <!-- We should be able to remove this extra property once Roslyn projects include the Sdk="Microsoft.NET.Sdk" attribute. -->
```

Comment:
So, I moved Roslyn over to using the SDK= syntax, but now that I better understand the implications of that, I'm not sure it's something we'd want (see https://github.com/KevinH-MS/roslyn/commit/c4bd8f86578618c255df653601c9ea7bd004f9fa).  There are a couple reasons:

1.  When using this syntax, you usually get the SDK props/targets from the latest dotnet SDK that you have installed on the machine (rather than the VS copy).  This may lead to unpredictable behavior across machines.
2.  Importing Microsoft.Common*.targets ***before*** setting any properties has subtle effects on the semantics of the Roslyn build.  We have not prioritized making Roslyn projects follow the conventions of the default project templates in the past.  While there would be some benefit to doing so, it's hard to quantify, and I'm not sure that outweighs the cost of the likely tail of issues (when considering that the alternative is a targeted, one-line fix).  I have adjusted the comment above so it does not suggest that Roslyn will/should be moving to the new syntax.